src/components/input/input.tsx

```diff
@@ -1164,6 +1169,7 @@ export class Input
               value={this.value}
               // eslint-disable-next-line react/jsx-sort-props
               ref={this.setChildElRef}
+              {...this.globalAttributes}
```

Comment:
@jcfranco can you elaborate on this one?

If a user sets a role on the host element, the same role will be set internally on this shadowRoot element?

Won't the nested roles conflict with each other? I'm not 100% on how screen readers read shadowRoot but it seems like a parent and a shadow child having the same role might be problematic?

Reply:
Also, if a user sets lang on the host it will be set on this element. That's not a problem but just seems redundant/not necessary.

Reply:
I think overriding the internal input's role is necessary to be able to implement the [combobox  patterns](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/#examples) from the ARIA Authoring Practices Guide:

> | Role	| Attribute	| Element |	Usage |
> |--------|--------|--------|--------|
> | combobox |		| input | 	Identifies the input element as a combobox.

> | Role	| Attribute	| Element |	Usage |
> |--------|--------|--------|--------|
> | combobox |		| input[type="text"] | 	Identifies the input element as a combobox.

Otherwise, the host will have one role and the actual input will have another one which might interfere with the screen reader interpretation. In my testing, VO reads and offers different options before and after making this change (compared to the example from https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-datepicker). I'll defer to @geospatialem on this one.

> Also, if a user sets lang on the host it will be set on this element. That's not a problem but just seems redundant/not necessary.

The util is configured to only watch `role` in this component (see https://github.com/Esri/calcite-components/pull/6805/files#diff-fe3d603d6390f87fdcdd3f40e34778ccdb07ff9edc52e440ad13c9571cfceadaR475 above).

Reply:
Agreed on the combobox role here. We did some testing with the light and shadow DOM to be sure, and both should be interpreted separately.